profiler/internal/cmemprof/pprof.go

```diff
@@ -6,13 +6,132 @@
 package cmemprof
 
 import (
+	"bytes"
 	"os"
 	"runtime"
+	"strconv"
 	"strings"
 
 	"github.com/google/pprof/profile"
 )
 
+var defaultMapping = &profile.Mapping{
+	ID:   1,
+	File: os.Args[0],
+}
+
+var mappings []*profile.Mapping
+
+func getMapping(addr uint64) *profile.Mapping {
+	for _, m := range mappings {
```

Comment:
🐌 How many `mappings` would you typically expect here? Could it become a perf issue for the profile creation?

Reply:
Yeah, that makes sense. I actually rewrote the parsing anyway to use a few convenience packages and make it a little simpler so it doesn't matter for this PR now. But for the next time we want to bring in some outside code, it looks like this repo has a [third-party licenses file](https://github.com/DataDog/dd-trace-go/blob/main/LICENSE-3rdparty.csv) where we can track the code we're borrowing.